### ISSUE TITLE
Redirect verification broken for OkHttp implementation

### ISSUE THREAD

**Issue by musketyr** [agorapulse/gru#48]
The OkHttp client is following the redirect and does not report the redirect URI

**Comment by akefirad** [comment]
I'm not sure if this is the same issue I'm having or not. I'm trying to write a test for an endpoint that return a redirect response. I don't want Gru to verify the redirection response not to follow it. Not sure how it's possible to it. Currently I disabled follow redirection on OkHttp client level and use `status` and `header` assertions instead of a simple [`redirect`](https://agorapulse.github.io/gru/#redirect):
```groovy
gru.verify {
    get("/redirect/foo/to/bar")

    expect {
        status 307
        // redirect "http://foo.bar/baz" <-- this fails with Expected URL http://foo.bar/baz but no redirection happened.
        header 'Location', "http://foo.bar/baz"
    }
}
```

**Comment by musketyr** [comment]
hi @akefirad, can you just let me know which version you are using? the `Http` client has been reworked recently. also a reproducer would be great to debug it.

**Comment by akefirad** [comment]
```kotlin
implementation(platform("com.agorapulse:gru-bom:2.1.0"))
implementation("com.agorapulse:gru-okhttp")
```
PS: I'm a little bit confused by Http vs OkHttp. The title of the issue says Http but the description says OkHttp. Are they not different things in the new version? (Or are you talking about the old version?)

**Comment by musketyr** [comment]
Yes, they were the same in in 1.x but separated in 2.x.

Can you also share how do you create the `Gru` instance? If you're using `OkHttp` it should something like

```
Gru gru = Gru.create(OkHttp.create(this))
```

if you use just `com.agorapulse:gru` then you get the default JDK based `Http` client.



I have updated the title to better match the current state.

**Comment by akefirad** [comment]
My actual code is like this (Spring Boot App + Spock):
```groovy
    @Bean
    @Scope("prototype")
    Gru gru(InjectionPoint injection) {
        def http = OkHttp.create(injection.getMember().getDeclaringClass()) {
            it.readTimeout(Duration.ZERO)
            it.followRedirects(false) // <- Note this, I don't want the redirect to be followed!
            it.addInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
        }
        def gru = Gru.create(http)
        return gru.prepare(serverUrl)
    }
```
Just to be clear, the issue is that I'm testing the redirection, I don't want Gru to follow the redirect.
Attaching a demo project (slightly different, but reproduces the same issue)
[demo.zip](https://github.com/agorapulse/gru/files/15355252/demo.zip)

**Comment by musketyr** [comment]
Fixed in  `2.1.1`
